// @aws-sdk/client-athena/dist-cjs/schemas/schemas_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataCatalogSummary$ = exports.DataCatalog$ = exports.Database$ = exports.CustomerContentEncryptionConfiguration$ = exports.CreateWorkGroupOutput$ = exports.CreateWorkGroupInput$ = exports.CreatePresignedNotebookUrlResponse$ = exports.CreatePresignedNotebookUrlRequest$ = exports.CreatePreparedStatementOutput$ = exports.CreatePreparedStatementInput$ = exports.CreateNotebookOutput$ = exports.CreateNotebookInput$ = exports.CreateNamedQueryOutput$ = exports.CreateNamedQueryInput$ = exports.CreateDataCatalogOutput$ = exports.CreateDataCatalogInput$ = exports.CreateCapacityReservationOutput$ = exports.CreateCapacityReservationInput$ = exports.ColumnInfo$ = exports.Column$ = exports.CloudWatchLoggingConfiguration$ = exports.Classification$ = exports.CapacityReservation$ = exports.CapacityAssignmentConfiguration$ = exports.CapacityAssignment$ = exports.CapacityAllocation$ = exports.CancelCapacityReservationOutput$ = exports.CancelCapacityReservationInput$ = exports.CalculationSummary$ = exports.CalculationStatus$ = exports.CalculationStatistics$ = exports.CalculationResult$ = exports.CalculationConfiguration$ = exports.BatchGetQueryExecutionOutput$ = exports.BatchGetQueryExecutionInput$ = exports.BatchGetPreparedStatementOutput$ = exports.BatchGetPreparedStatementInput$ = exports.BatchGetNamedQueryOutput$ = exports.BatchGetNamedQueryInput$ = exports.AthenaError$ = exports.ApplicationDPUSizes$ = exports.AclConfiguration$ = exports.errorTypeRegistries = exports.TooManyRequestsException$ = exports.SessionAlreadyExistsException$ = exports.ResourceNotFoundException$ = exports.MetadataException$ = exports.InvalidRequestException$ = exports.InternalServerException$ = exports.AthenaServiceException$ = void 0;
exports.GetSessionEndpointResponse$ = exports.GetSessionEndpointRequest$ = exports.GetResourceDashboardResponse$ = exports.GetResourceDashboardRequest$ = exports.GetQueryRuntimeStatisticsOutput$ = exports.GetQueryRuntimeStatisticsInput$ = exports.GetQueryResultsOutput$ = exports.GetQueryResultsInput$ = exports.GetQueryExecutionOutput$ = exports.GetQueryExecutionInput$ = exports.GetPreparedStatementOutput$ = exports.GetPreparedStatementInput$ = exports.GetNotebookMetadataOutput$ = exports.GetNotebookMetadataInput$ = exports.GetNamedQueryOutput$ = exports.GetNamedQueryInput$ = exports.GetDataCatalogOutput$ = exports.GetDataCatalogInput$ = exports.GetDatabaseOutput$ = exports.GetDatabaseInput$ = exports.GetCapacityReservationOutput$ = exports.GetCapacityReservationInput$ = exports.GetCapacityAssignmentConfigurationOutput$ = exports.GetCapacityAssignmentConfigurationInput$ = exports.GetCalculationExecutionStatusResponse$ = exports.GetCalculationExecutionStatusRequest$ = exports.GetCalculationExecutionResponse$ = exports.GetCalculationExecutionRequest$ = exports.GetCalculationExecutionCodeResponse$ = exports.GetCalculationExecutionCodeRequest$ = exports.FilterDefinition$ = exports.ExportNotebookOutput$ = exports.ExportNotebookInput$ = exports.ExecutorsSummary$ = exports.EngineVersion$ = exports.EngineConfiguration$ = exports.EncryptionConfiguration$ = exports.DeleteWorkGroupOutput$ = exports.DeleteWorkGroupInput$ = exports.DeletePreparedStatementOutput$ = exports.DeletePreparedStatementInput$ = exports.DeleteNotebookOutput$ = exports.DeleteNotebookInput$ = exports.DeleteNamedQueryOutput$ = exports.DeleteNamedQueryInput$ = exports.DeleteDataCatalogOutput$ = exports.DeleteDataCatalogInput$ = exports.DeleteCapacityReservationOutput$ = exports.DeleteCapacityReservationInput$ = exports.Datum$ = void 0;
exports.NotebookMetadata$ = exports.NamedQuery$ = exports.MonitoringConfiguration$ = exports.ManagedQueryResultsEncryptionConfiguration$ = exports.ManagedQueryResultsConfigurationUpdates$ = exports.ManagedQueryResultsConfiguration$ = exports.ManagedLoggingConfiguration$ = exports.ListWorkGroupsOutput$ = exports.ListWorkGroupsInput$ = exports.ListTagsForResourceOutput$ = exports.ListTagsForResourceInput$ = exports.ListTableMetadataOutput$ = exports.ListTableMetadataInput$ = exports.ListSessionsResponse$ = exports.ListSessionsRequest$ = exports.ListQueryExecutionsOutput$ = exports.ListQueryExecutionsInput$ = exports.ListPreparedStatementsOutput$ = exports.ListPreparedStatementsInput$ = exports.ListNotebookSessionsResponse$ = exports.ListNotebookSessionsRequest$ = exports.ListNotebookMetadataOutput$ = exports.ListNotebookMetadataInput$ = exports.ListNamedQueriesOutput$ = exports.ListNamedQueriesInput$ = exports.ListExecutorsResponse$ = exports.ListExecutorsRequest$ = exports.ListEngineVersionsOutput$ = exports.ListEngineVersionsInput$ = exports.ListDataCatalogsOutput$ = exports.ListDataCatalogsInput$ = exports.ListDatabasesOutput$ = exports.ListDatabasesInput$ = exports.ListCapacityReservationsOutput$ = exports.ListCapacityReservationsInput$ = exports.ListCalculationExecutionsResponse$ = exports.ListCalculationExecutionsRequest$ = exports.ListApplicationDPUSizesOutput$ = exports.ListApplicationDPUSizesInput$ = exports.ImportNotebookOutput$ = exports.ImportNotebookInput$ = exports.IdentityCenterConfiguration$ = exports.GetWorkGroupOutput$ = exports.GetWorkGroupInput$ = exports.GetTableMetadataOutput$ = exports.GetTableMetadataInput$ = exports.GetSessionStatusResponse$ = exports.GetSessionStatusRequest$ = exports.GetSessionResponse$ = exports.GetSessionRequest$ = void 0;
exports.UpdateCapacityReservationInput$ = exports.UntagResourceOutput$ = exports.UntagResourceInput$ = exports.UnprocessedQueryExecutionId$ = exports.UnprocessedPreparedStatementName$ = exports.UnprocessedNamedQueryId$ = exports.TerminateSessionResponse$ = exports.TerminateSessionRequest$ = exports.TagResourceOutput$ = exports.TagResourceInput$ = exports.Tag$ = exports.TableMetadata$ = exports.StopQueryExecutionOutput$ = exports.StopQueryExecutionInput$ = exports.StopCalculationExecutionResponse$ = exports.StopCalculationExecutionRequest$ = exports.StartSessionResponse$ = exports.StartSessionRequest$ = exports.StartQueryExecutionOutput$ = exports.StartQueryExecutionInput$ = exports.StartCalculationExecutionResponse$ = exports.StartCalculationExecutionRequest$ = exports.SessionSummary$ = exports.SessionStatus$ = exports.SessionStatistics$ = exports.SessionConfiguration$ = exports.S3LoggingConfiguration$ = exports.Row$ = exports.ResultSetMetadata$ = exports.ResultSet$ = exports.ResultReuseInformation$ = exports.ResultReuseConfiguration$ = exports.ResultReuseByAgeConfiguration$ = exports.ResultConfigurationUpdates$ = exports.ResultConfiguration$ = exports.QueryStagePlanNode$ = exports.QueryStage$ = exports.QueryRuntimeStatisticsTimeline$ = exports.QueryRuntimeStatisticsRows$ = exports.QueryRuntimeStatistics$ = exports.QueryResultsS3AccessGrantsConfiguration$ = exports.QueryExecutionStatus$ = exports.QueryExecutionStatistics$ = exports.QueryExecutionContext$ = exports.QueryExecution$ = exports.PutCapacityAssignmentConfigurationOutput$ = exports.PutCapacityAssignmentConfigurationInput$ = exports.PreparedStatementSummary$ = exports.PreparedStatement$ = exports.NotebookSessionSummary$ = void 0;
exports.GetSession$ = exports.GetResourceDashboard$ = exports.GetQueryRuntimeStatistics$ = exports.GetQueryResults$ = exports.GetQueryExecution$ = exports.GetPreparedStatement$ = exports.GetNotebookMetadata$ = exports.GetNamedQuery$ = exports.GetDataCatalog$ = exports.GetDatabase$ = exports.GetCapacityReservation$ = exports.GetCapacityAssignmentConfiguration$ = exports.GetCalculationExecutionStatus$ = exports.GetCalculationExecutionCode$ = exports.GetCalculationExecution$ = exports.ExportNotebook$ = exports.DeleteWorkGroup$ = exports.DeletePreparedStatement$ = exports.DeleteNotebook$ = exports.DeleteNamedQuery$ = exports.DeleteDataCatalog$ = exports.DeleteCapacityReservation$ = exports.CreateWorkGroup$ = exports.CreatePresignedNotebookUrl$ = exports.CreatePreparedStatement$ = exports.CreateNotebook$ = exports.CreateNamedQuery$ = exports.CreateDataCatalog$ = exports.CreateCapacityReservation$ = exports.CancelCapacityReservation$ = exports.BatchGetQueryExecution$ = exports.BatchGetPreparedStatement$ = exports.BatchGetNamedQuery$ = exports.WorkGroupSummary$ = exports.WorkGroupConfigurationUpdates$ = exports.WorkGroupConfiguration$ = exports.WorkGroup$ = exports.UpdateWorkGroupOutput$ = exports.UpdateWorkGroupInput$ = exports.UpdatePreparedStatementOutput$ = exports.UpdatePreparedStatementInput$ = exports.UpdateNotebookOutput$ = exports.UpdateNotebookMetadataOutput$ = exports.UpdateNotebookMetadataInput$ = exports.UpdateNotebookInput$ = exports.UpdateNamedQueryOutput$ = exports.UpdateNamedQueryInput$ = exports.UpdateDataCatalogOutput$ = exports.UpdateDataCatalogInput$ = exports.UpdateCapacityReservationOutput$ = void 0;
exports.UpdateWorkGroup$ = exports.UpdatePreparedStatement$ = exports.UpdateNotebookMetadata$ = exports.UpdateNotebook$ = exports.UpdateNamedQuery$ = exports.UpdateDataCatalog$ = exports.UpdateCapacityReservation$ = exports.UntagResource$ = exports.TerminateSession$ = exports.TagResource$ = exports.StopQueryExecution$ = exports.StopCalculationExecution$ = exports.StartSession$ = exports.StartQueryExecution$ = exports.StartCalculationExecution$ = exports.PutCapacityAssignmentConfiguration$ = exports.ListWorkGroups$ = exports.ListTagsForResource$ = exports.ListTableMetadata$ = exports.ListSessions$ = exports.ListQueryExecutions$ = exports.ListPreparedStatements$ = exports.ListNotebookSessions$ = exports.ListNotebookMetadata$ = exports.ListNamedQueries$ = exports.ListExecutors$ = exports.ListEngineVersions$ = exports.ListDataCatalogs$ = exports.ListDatabases$ = exports.ListCapacityReservations$ = exports.ListCalculationExecutions$ = exports.ListApplicationDPUSizes$ = exports.ImportNotebook$ = exports.GetWorkGroup$ = exports.GetTableMetadata$ = exports.GetSessionStatus$ = exports.GetSessionEndpoint$ = void 0;
const _AC = "AclConfiguration";
const _ACd = "AdditionalConfigs";
const _ACdd = "AdditionalConfiguration";
const _AD = "AllocatedDpus";
const _ADPUS = "ApplicationDPUSizes";
const _ADPUSL = "ApplicationDPUSizesList";
const _AE = "AthenaError";
const _AEC = "AthenaErrorCode";
const _ARI = "ApplicationRuntimeId";
const _AT = "AuthToken";
const _ATET = "AuthTokenExpirationTime";
const _ATu = "AuthenticationType";
const _BGNQ = "BatchGetNamedQuery";
const _BGNQI = "BatchGetNamedQueryInput";
const _BGNQO = "BatchGetNamedQueryOutput";
const _BGPS = "BatchGetPreparedStatement";
const _BGPSI = "BatchGetPreparedStatementInput";
const _BGPSO = "BatchGetPreparedStatementOutput";
const _BGQE = "BatchGetQueryExecution";
const _BGQEI = "BatchGetQueryExecutionInput";
const _BGQEO = "BatchGetQueryExecutionOutput";
const _BSCPQ = "BytesScannedCutoffPerQuery";
const _C = "Classification";
const _CA = "CapacityAllocation";
const _CAC = "CapacityAssignmentConfiguration";
const _CAL = "CapacityAssignmentsList";
const _CAa = "CapacityAssignment";
const _CAap = "CapacityAssignments";
const _CB = "CodeBlock";
const _CC = "CalculationConfiguration";
const _CCEC = "CustomerContentEncryptionConfiguration";
const _CCR = "CancelCapacityReservation";
const _CCRI = "CancelCapacityReservationInput";
const _CCRIr = "CreateCapacityReservationInput";
const _CCRO = "CancelCapacityReservationOutput";
const _CCROr = "CreateCapacityReservationOutput";
const _CCRr = "CreateCapacityReservation";
const _CDC = "CreateDataCatalog";
const _CDCI = "CreateDataCatalogInput";
const _CDCO = "CreateDataCatalogOutput";
const _CDS = "CoordinatorDpuSize";
const _CDT = "CompletionDateTime";
const _CEI = "CalculationExecutionId";
const _CI = "ColumnInfo";
const _CIL = "ColumnInfoList";
const _CL = "CalculationsList";
const _CLl = "ClassificationList";
const _CLo = "ColumnList";
const _CN = "CatalogName";
const _CNI = "CreateNotebookInput";
const _CNO = "CreateNotebookOutput";
const _CNQ = "CreateNamedQuery";
const _CNQI = "CreateNamedQueryInput";
const _CNQO = "CreateNamedQueryOutput";
const _CNr = "CreateNotebook";
const _CPNU = "CreatePresignedNotebookUrl";
const _CPNUR = "CreatePresignedNotebookUrlRequest";
const _CPNURr = "CreatePresignedNotebookUrlResponse";
const _CPS = "CreatePreparedStatement";
const _CPSI = "CreatePreparedStatementInput";
const _CPSO = "CreatePreparedStatementOutput";
const _CR = "CalculationResult";
const _CRL = "CapacityReservationsList";
const _CRN = "CapacityReservationName";
const _CRT = "ClientRequestToken";
const _CRa = "CapacityReservation";
const _CRap = "CapacityReservations";
const _CS = "CalculationStatistics";
const _CSa = "CalculationStatus";
const _CSal = "CalculationSummary";
const _CSas = "CaseSensitive";
const _CT = "CreationTime";
const _CTo = "ConnectionType";
const _CTr = "CreateTime";
const _CU = "ConfigurationUpdates";
const _CULP = "CreateUserLevelPrefix";
const _CWG = "CreateWorkGroup";
const _CWGI = "CreateWorkGroupInput";
const _CWGO = "CreateWorkGroupOutput";
const _CWGT = "CopyWorkGroupTags";
const _CWLC = "CloudWatchLoggingConfiguration";
const _Ca = "Calculations";
const _Cat = "Catalog";
const _Ch = "Children";
const _Cl = "Classifications";
const _Co = "Column";
const _Col = "Columns";
const _Com = "Comment";
const _Con = "Configuration";
const _D = "Description";
const _DC = "DataCatalog";
const _DCO = "DeleteCatalogOnly";
const _DCR = "DeleteCapacityReservation";
const _DCRI = "DeleteCapacityReservationInput";
const _DCRO = "DeleteCapacityReservationOutput";
const _DCS = "DataCatalogSummary";
const _DCSL = "DataCatalogSummaryList";
const _DCSa = "DataCatalogsSummary";
const _DCp = "DpuCount";
const _DDC = "DeleteDataCatalog";
const _DDCI = "DeleteDataCatalogInput";
const _DDCO = "DeleteDataCatalogOutput";
const _DEDS = "DefaultExecutorDpuSize";
const _DEIM = "DpuExecutionInMillis";
const _DL = "DatabaseList";
const _DML = "DataManifestLocation";
const _DN = "DatabaseName";
const _DNI = "DeleteNotebookInput";
const _DNO = "DeleteNotebookOutput";
const _DNQ = "DeleteNamedQuery";
const _DNQI = "DeleteNamedQueryInput";
const _DNQO = "DeleteNamedQueryOutput";
const _DNe = "DeleteNotebook";
const _DPS = "DeletePreparedStatement";
const _DPSI = "DeletePreparedStatementInput";
const _DPSO = "DeletePreparedStatementOutput";
const _DSIB = "DataScannedInBytes";
const _DWG = "DeleteWorkGroup";
const _DWGI = "DeleteWorkGroupInput";
const _DWGO = "DeleteWorkGroupOutput";
const _Da = "Database";
const _Dat = "Datum";
const _Data = "Data";
const _E = "Enabled";
const _EBO = "ExpectedBucketOwner";
const _EC = "ErrorCategory";
const _ECn = "EncryptionConfiguration";
const _ECng = "EngineConfiguration";
const _ECr = "ErrorCode";
const _EDT = "EndDateTime";
const _EETIM = "EngineExecutionTimeInMillis";
const _EEV = "EffectiveEngineVersion";
const _EI = "ExecutorId";
const _EIC = "EnableIdentityCenter";
const _EM = "ErrorMessage";
const _EMEC = "EnableMinimumEncryptionConfiguration";
const _EN = "ExportNotebook";
const _ENI = "ExportNotebookInput";
const _ENO = "ExportNotebookOutput";
const _EO = "EncryptionOption";
const _EP = "ExecutionParameters";
const _ER = "ExecutionRole";
const _ES = "ExecutorsSummary";
const _ESAG = "EnableS3AccessGrants";
const _ESF = "ExecutorStateFilter";
const _ESL = "ExecutorsSummaryList";
const _ESx = "ExecutorState";
const _ESxe = "ExecutorSize";
const _ET = "ErrorType";
const _ETx = "ExecutorType";
const _ETxe = "ExecutionTime";
const _EU = "EndpointUrl";
const _EV = "EngineVersion";
const _EVL = "EngineVersionsList";
const _EVn = "EngineVersions";
const _EWGC = "EnforceWorkGroupConfiguration";
const _Er = "Error";
const _Ex = "Expression";
const _F = "Filters";
const _FD = "FilterDefinition";
const _GCAC = "GetCapacityAssignmentConfiguration";
const _GCACI = "GetCapacityAssignmentConfigurationInput";
const _GCACO = "GetCapacityAssignmentConfigurationOutput";
const _GCE = "GetCalculationExecution";
const _GCEC = "GetCalculationExecutionCode";
const _GCECR = "GetCalculationExecutionCodeRequest";
const _GCECRe = "GetCalculationExecutionCodeResponse";
const _GCER = "GetCalculationExecutionRequest";
const _GCERe = "GetCalculationExecutionResponse";
const _GCES = "GetCalculationExecutionStatus";
const _GCESR = "GetCalculationExecutionStatusRequest";
const _GCESRe = "GetCalculationExecutionStatusResponse";
const _GCR = "GetCapacityReservation";
const _GCRI = "GetCapacityReservationInput";
const _GCRO = "GetCapacityReservationOutput";
const _GD = "GetDatabase";
const _GDC = "GetDataCatalog";
const _GDCI = "GetDataCatalogInput";
const _GDCO = "GetDataCatalogOutput";
const _GDI = "GetDatabaseInput";
const _GDO = "GetDatabaseOutput";
const _GNM = "GetNotebookMetadata";
const _GNMI = "GetNotebookMetadataInput";
const _GNMO = "GetNotebookMetadataOutput";
const _GNQ = "GetNamedQuery";
const _GNQI = "GetNamedQueryInput";
const _GNQO = "GetNamedQueryOutput";
const _GPS = "GetPreparedStatement";
const _GPSI = "GetPreparedStatementInput";
const _GPSO = "GetPreparedStatementOutput";
const _GQE = "GetQueryExecution";
const _GQEI = "GetQueryExecutionInput";
const _GQEO = "GetQueryExecutionOutput";
const _GQR = "GetQueryResults";
const _GQRI = "GetQueryResultsInput";
const _GQRO = "GetQueryResultsOutput";
const _GQRS = "GetQueryRuntimeStatistics";
const _GQRSI = "GetQueryRuntimeStatisticsInput";
const _GQRSO = "GetQueryRuntimeStatisticsOutput";
const _GRD = "GetResourceDashboard";
const _GRDR = "GetResourceDashboardRequest";
const _GRDRe = "GetResourceDashboardResponse";
const _GS = "GetSession";
const _GSE = "GetSessionEndpoint";
const _GSER = "GetSessionEndpointRequest";
const _GSERe = "GetSessionEndpointResponse";
const _GSR = "GetSessionRequest";
const _GSRe = "GetSessionResponse";
const _GSS = "GetSessionStatus";
const _GSSR = "GetSessionStatusRequest";
const _GSSRe = "GetSessionStatusResponse";
const _GTM = "GetTableMetadata";
const _GTMI = "GetTableMetadataInput";
const _GTMO = "GetTableMetadataOutput";
const _GWG = "GetWorkGroup";
const _GWGI = "GetWorkGroupInput";
const _GWGO = "GetWorkGroupOutput";
const _I = "Identifier";
const _IB = "InputBytes";
const _ICAA = "IdentityCenterApplicationArn";
const _ICC = "IdentityCenterConfiguration";
const _ICIA = "IdentityCenterInstanceArn";
const _IN = "ImportNotebook";
const _INI = "ImportNotebookInput";
const _INO = "ImportNotebookOutput";
const _IR = "InputRows";
const _IRE = "InvalidRequestException";
const _ISDT = "IdleSinceDateTime";
const _ISE = "InternalServerException";
const _ITS = "IdleTimeoutSeconds";
const _K = "Key";
const _KK = "KmsKey";
const _L = "Label";
const _LA = "LastAllocation";
const _LADPUS = "ListApplicationDPUSizes";
const _LADPUSI = "ListApplicationDPUSizesInput";
const _LADPUSO = "ListApplicationDPUSizesOutput";
const _LAT = "LastAccessTime";
const _LCE = "ListCalculationExecutions";
const _LCER = "ListCalculationExecutionsRequest";
const _LCERi = "ListCalculationExecutionsResponse";
const _LCR = "ListCapacityReservations";
const _LCRI = "ListCapacityReservationsInput";
const _LCRO = "ListCapacityReservationsOutput";
const _LD = "ListDatabases";
const _LDC = "ListDataCatalogs";
const _LDCI = "ListDataCatalogsInput";
const _LDCO = "ListDataCatalogsOutput";
const _LDI = "ListDatabasesInput";
const _LDO = "ListDatabasesOutput";
const _LE = "ListExecutors";
const _LER = "ListExecutorsRequest";
const _LERi = "ListExecutorsResponse";
const _LEV = "ListEngineVersions";
const _LEVI = "ListEngineVersionsInput";
const _LEVO = "ListEngineVersionsOutput";
const _LG = "LogGroup";
const _LL = "LogLocation";
const _LMDT = "LastModifiedDateTime";
const _LMT = "LastModifiedTime";
const _LNM = "ListNotebookMetadata";
const _LNMI = "ListNotebookMetadataInput";
const _LNMO = "ListNotebookMetadataOutput";
const _LNQ = "ListNamedQueries";
const _LNQI = "ListNamedQueriesInput";
const _LNQO = "ListNamedQueriesOutput";
const _LNS = "ListNotebookSessions";
const _LNSR = "ListNotebookSessionsRequest";
const _LNSRi = "ListNotebookSessionsResponse";
const _LPS = "ListPreparedStatements";
const _LPSI = "ListPreparedStatementsInput";
const _LPSO = "ListPreparedStatementsOutput";
const _LQE = "ListQueryExecutions";
const _LQEI = "ListQueryExecutionsInput";
const _LQEO = "ListQueryExecutionsOutput";
const _LS = "ListSessions";
const _LSAT = "LastSuccessfulAllocationTime";
const _LSNP = "LogStreamNamePrefix";
const _LSR = "ListSessionsRequest";
const _LSRi = "ListSessionsResponse";
const _LT = "LogTypes";
const _LTFR = "ListTagsForResource";
const _LTFRI = "ListTagsForResourceInput";
const _LTFRO = "ListTagsForResourceOutput";
const _LTM = "LogTypesMap";
const _LTMI = "ListTableMetadataInput";
const _LTMO = "ListTableMetadataOutput";
const _LTMi = "ListTableMetadata";
const _LWG = "ListWorkGroups";
const _LWGI = "ListWorkGroupsInput";
const _LWGO = "ListWorkGroupsOutput";
const _M = "Message";
const _MAIM = "MaxAgeInMinutes";
const _MC = "MonitoringConfiguration";
const _MCD = "MaxConcurrentDpus";
const _ME = "MetadataException";
const _MLC = "ManagedLoggingConfiguration";
const _MQRC = "ManagedQueryResultsConfiguration";
const _MQRCU = "ManagedQueryResultsConfigurationUpdates";
const _MQREC = "ManagedQueryResultsEncryptionConfiguration";
const _MR = "MaxResults";
const _N = "Name";
const _NI = "NotebookId";
const _NM = "NotebookMetadata";
const _NMA = "NotebookMetadataArray";
const _NML = "NotebookMetadataList";
const _NQ = "NamedQueries";
const _NQI = "NamedQueryIds";
const _NQIa = "NamedQueryId";
const _NQL = "NamedQueryList";
const _NQa = "NamedQuery";
const _NSL = "NotebookSessionsList";
const _NSLU = "NotebookS3LocationUri";
const _NSS = "NotebookSessionSummary";
const _NT = "NextToken";
const _NU = "NotebookUrl";
const _NV = "NotebookVersion";
const _Nu = "Nullable";
const _OB = "OutputBytes";
const _OL = "OutputLocation";
const _OR = "OutputRows";
const _OS = "OutputStage";
const _P = "Progress";
const _PCAC = "PutCapacityAssignmentConfiguration";
const _PCACI = "PutCapacityAssignmentConfigurationInput";
const _PCACO = "PutCapacityAssignmentConfigurationOutput";
const _PCWME = "PublishCloudWatchMetricsEnabled";
const _PK = "PartitionKeys";
const _PS = "PreparedStatements";
const _PSDL = "PreparedStatementDetailsList";
const _PSL = "PreparedStatementsList";
const _PSN = "PreparedStatementNames";
const _PSS = "PreparedStatementSummary";
const _PSr = "PreparedStatement";
const _Pa = "Parameters";
const _Pay = "Payload";
const _Pr = "Properties";
const _Pre = "Precision";
const _Q = "Query";
const _QE = "QueryExecutions";
const _QEC = "QueryExecutionContext";
const _QEI = "QueryExecutionIds";
const _QEIu = "QueryExecutionId";
const _QEL = "QueryExecutionList";
const _QES = "QueryExecutionStatistics";
const _QESu = "QueryExecutionStatus";
const _QEu = "QueryExecution";
const _QPTIM = "QueryPlanningTimeInMillis";
const _QQTIM = "QueryQueueTimeInMillis";
const _QRS = "QueryRuntimeStatistics";
const _QRSAGC = "QueryResultsS3AccessGrantsConfiguration";
const _QRSR = "QueryRuntimeStatisticsRows";
const _QRST = "QueryRuntimeStatisticsTimeline";
const _QRT = "QueryResultType";
const _QS = "QueryString";
const _QSP = "QueryStagePlan";
const _QSPN = "QueryStagePlanNode";
const _QSPNu = "QueryStagePlanNodes";
const _QSu = "QueryStatement";
const _QSue = "QueryStage";
const _QSuer = "QueryStages";
const _R = "Reason";
const _RAC = "RemoveAclConfiguration";
const _RARN = "ResourceARN";
const _RBSCPQ = "RemoveBytesScannedCutoffPerQuery";
const _RC = "ResultConfiguration";
const _RCCEC = "RemoveCustomerContentEncryptionConfiguration";
const _RCT = "RequestCompletionTime";
const _RCU = "ResultConfigurationUpdates";
const _RDO = "RecursiveDeleteOption";
const _REBO = "RemoveExpectedBucketOwner";
const _REC = "RemoveEncryptionConfiguration";
const _RL = "RowList";
const _RN = "ResourceName";
const _RNFE = "ResourceNotFoundException";
const _ROL = "RemoveOutputLocation";
const _RPE = "RequesterPaysEnabled";
const _RPR = "ReusedPreviousResult";
const _RRBAC = "ResultReuseByAgeConfiguration";
const _RRC = "ResultReuseConfiguration";
const _RRI = "ResultReuseInformation";
const _RS = "ResultSet";
const _RSM = "ResultSetMetadata";
const _RSU = "ResultS3Uri";
const _RSe = "RemoteSources";
const _RT = "ResultType";
const _RTe = "RequestTime";
const _Re = "Retryable";
const _Res = "Result";
const _Ro = "Rows";
const _Row = "Row";
const _S = "State";
const _SAEE = "SessionAlreadyExistsException";
const _SAO = "S3AclOption";
const _SC = "SessionConfiguration";
const _SCE = "StartCalculationExecution";
const _SCER = "StartCalculationExecutionRequest";
const _SCERt = "StartCalculationExecutionResponse";
const _SCERto = "StopCalculationExecutionRequest";
const _SCERtop = "StopCalculationExecutionResponse";
const _SCEt = "StopCalculationExecution";
const _SCR = "StateChangeReason";
const _SDPUS = "SupportedDPUSizes";
const _SDT = "SubmissionDateTime";
const _SDTt = "StartDateTime";
const _SESU = "StdErrorS3Uri";
const _SEV = "SelectedEngineVersion";
const _SF = "StateFilter";
const _SI = "SessionId";
const _SITIM = "SessionIdleTimeoutInMinutes";
const _SIt = "StageId";
const _SL = "SessionsList";
const _SLC = "S3LoggingConfiguration";
const _SM = "StatusMessage";
const _SN = "SchemaName";
const _SNt = "StatementName";
const _SOSU = "StdOutS3Uri";
const _SP = "SparkProperties";
const _SPPTIM = "ServicePreProcessingTimeInMillis";
const _SPTIM = "ServiceProcessingTimeInMillis";
const _SQE = "StartQueryExecution";
const _SQEI = "StartQueryExecutionInput";
const _SQEIt = "StopQueryExecutionInput";
const _SQEO = "StartQueryExecutionOutput";
const _SQEOt = "StopQueryExecutionOutput";
const _SQEt = "StopQueryExecution";
const _SS = "SubStages";
const _SSR = "StartSessionRequest";
const _SSRt = "StartSessionResponse";
const _SSe = "SessionStatistics";
const _SSes = "SessionStatus";
const _SSess = "SessionSummary";
const _SSt = "StartSession";
const _ST = "StatementType";
const _STu = "SubstatementType";
const _Sc = "Scale";
const _Se = "Sessions";
const _St = "Status";
const _Sta = "Statistics";
const _T = "Type";
const _TD = "TargetDpus";
const _TDT = "TerminationDateTime";
const _TETIM = "TotalExecutionTimeInMillis";
const _TK = "TagKeys";
const _TL = "TagList";
const _TM = "TableMetadata";
const _TML = "TableMetadataList";
const _TMRE = "TooManyRequestsException";
const _TN = "TableName";
const _TR = "TagResource";
const _TRI = "TagResourceInput";
const _TRO = "TagResourceOutput";
const _TS = "TerminateSession";
const _TSR = "TerminateSessionRequest";
const _TSRe = "TerminateSessionResponse";
const _TT = "TableType";
const _Ta = "Tags";
const _Tag = "Tag";
const _Ti = "Timeline";
const _U = "Url";
const _UC = "UpdateCount";
const _UCR = "UpdateCapacityReservation";
const _UCRI = "UpdateCapacityReservationInput";
const _UCRO = "UpdateCapacityReservationOutput";
const _UDC = "UpdateDataCatalog";
const _UDCI = "UpdateDataCatalogInput";
const _UDCO = "UpdateDataCatalogOutput";
const _UN = "UpdateNotebook";
const _UNI = "UpdateNotebookInput";
const _UNM = "UpdateNotebookMetadata";
const _UNMI = "UpdateNotebookMetadataInput";
const _UNMO = "UpdateNotebookMetadataOutput";
const _UNO = "UpdateNotebookOutput";
const _UNQ = "UpdateNamedQuery";
const _UNQI = "UnprocessedNamedQueryIds";
const _UNQIL = "UnprocessedNamedQueryIdList";
const _UNQIn = "UnprocessedNamedQueryId";
const _UNQIp = "UpdateNamedQueryInput";
const _UNQO = "UpdateNamedQueryOutput";
const _UPS = "UpdatePreparedStatement";
const _UPSI = "UpdatePreparedStatementInput";
const _UPSN = "UnprocessedPreparedStatementNames";
const _UPSNL = "UnprocessedPreparedStatementNameList";
const _UPSNn = "UnprocessedPreparedStatementName";
const _UPSO = "UpdatePreparedStatementOutput";
const _UQEI = "UnprocessedQueryExecutionIds";
const _UQEIL = "UnprocessedQueryExecutionIdList";
const _UQEIn = "UnprocessedQueryExecutionId";
const _UR = "UntagResource";
const _URI = "UntagResourceInput";
const _URO = "UntagResourceOutput";
const _UWG = "UpdateWorkGroup";
const _UWGI = "UpdateWorkGroupInput";
const _UWGO = "UpdateWorkGroupOutput";
const _V = "Value";
const _VCV = "VarCharValue";
const _WD = "WorkingDirectory";
const _WG = "WorkGroup";
const _WGC = "WorkGroupConfiguration";
const _WGCU = "WorkGroupConfigurationUpdates";
const _WGL = "WorkGroupsList";
const _WGN = "WorkGroupNames";
const _WGNo = "WorkGroupName";
const _WGS = "WorkGroupSummary";
const _WGo = "WorkGroups";
const _c = "client";
const _dL = "datumList";
const _e = "error";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.athena";
const _se = "server";
const n0 = "com.amazonaws.athena";
const schema_1 = require("@smithy/core/schema");
const AthenaServiceException_1 = require("../models/AthenaServiceException");
const errors_1 = require("../models/errors");
const _s_registry = schema_1.TypeRegistry.for(_s);
exports.AthenaServiceException$ = [-3, _s, "AthenaServiceException", 0, [], []];
_s_registry.registerError(exports.AthenaServiceException$, AthenaServiceException_1.AthenaServiceException);
const n0_registry = schema_1.TypeRegistry.for(n0);
exports.InternalServerException$ = [-3, n0, _ISE,
    { [_e]: _se },
    [_M],
    [0]
];
n0_registry.registerError(exports.InternalServerException$, errors_1.InternalServerException);
exports.InvalidRequestException$ = [-3, n0, _IRE,
    { [_e]: _c },
    [_AEC, _M],
    [0, 0]
];
n0_registry.registerError(exports.InvalidRequestException$, errors_1.InvalidRequestException);
exports.MetadataException$ = [-3, n0, _ME,
    { [_e]: _c },
    [_M],
    [0]
];
n0_registry.registerError(exports.MetadataException$, errors_1.MetadataException);
exports.ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c },
    [_M, _RN],
    [0, 0]
];
n0_registry.registerError(exports.ResourceNotFoundException$, errors_1.ResourceNotFoundException);
exports.SessionAlreadyExistsException$ = [-3, n0, _SAEE,
    { [_e]: _c },
    [_M],
    [0]
];
n0_registry.registerError(exports.SessionAlreadyExistsException$, errors_1.SessionAlreadyExistsException);
exports.TooManyRequestsException$ = [-3, n0, _TMRE,
    { [_e]: _c },
    [_M, _R],
    [0, 0]
];
n0_registry.registerError(exports.TooManyRequestsException$, errors_1.TooManyRequestsException);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
exports.AclConfiguration$ = [3, n0, _AC,
    0,
    [_SAO],
    [0], 1
];
exports.ApplicationDPUSizes$ = [3, n0, _ADPUS,
    0,
    [_ARI, _SDPUS],
    [0, 64 | 1]
];
exports.AthenaError$ = [3, n0, _AE,
    0,
    [_EC, _ET, _Re, _EM],
    [1, 1, 2, 0]
];
exports.BatchGetNamedQueryInput$ = [3, n0, _BGNQI,
    0,
    [_NQI],
    [64 | 0], 1
];
exports.BatchGetNamedQueryOutput$ = [3, n0, _BGNQO,
    0,
    [_NQ, _UNQI],
    [() => NamedQueryList, () => UnprocessedNamedQueryIdList]
];
exports.BatchGetPreparedStatementInput$ = [3, n0, _BGPSI,
    0,
    [_PSN, _WG],
    [64 | 0, 0], 2
];
exports.BatchGetPreparedStatementOutput$ = [3, n0, _BGPSO,
    0,
    [_PS, _UPSN],
    [() => PreparedStatementDetailsList, () => UnprocessedPreparedStatementNameList]
];
exports.BatchGetQueryExecutionInput$ = [3, n0, _BGQEI,
    0,
    [_QEI],
    [64 | 0], 1
];
exports.BatchGetQueryExecutionOutput$ = [3, n0, _BGQEO,
    0,
    [_QE, _UQEI],
    [() => QueryExecutionList, () => UnprocessedQueryExecutionIdList]
];
exports.CalculationConfiguration$ = [3, n0, _CC,
    0,
    [_CB],
    [0]
];
exports.CalculationResult$ = [3, n0, _CR,
    0,
    [_SOSU, _SESU, _RSU, _RT],
    [0, 0, 0, 0]
];
exports.CalculationStatistics$ = [3, n0, _CS,
    0,
    [_DEIM, _P],
    [1, 0]
];
exports.CalculationStatus$ = [3, n0, _CSa,
    0,
    [_SDT, _CDT, _S, _SCR],
    [4, 4, 0, 0]
];
exports.CalculationSummary$ = [3, n0, _CSal,
    0,
    [_CEI, _D, _St],
    [0, 0, () => exports.CalculationStatus$]
];
exports.CancelCapacityReservationInput$ = [3, n0, _CCRI,
    0,
    [_N],
    [0], 1
];
exports.CancelCapacityReservationOutput$ = [3, n0, _CCRO,
    0,
    [],
    []
];
exports.CapacityAllocation$ = [3, n0, _CA,
    0,
    [_St, _RTe, _SM, _RCT],
    [0, 4, 0, 4], 2
];
exports.CapacityAssignment$ = [3, n0, _CAa,
    0,
    [_WGN],
    [64 | 0]
];
exports.CapacityAssignmentConfiguration$ = [3, n0, _CAC,
    0,
    [_CRN, _CAap],
    [0, () => CapacityAssignmentsList]
];
exports.CapacityReservation$ = [3, n0, _CRa,
    0,
    [_N, _St, _TD, _AD, _CT, _LA, _LSAT],
    [0, 0, 1, 1, 4, () => exports.CapacityAllocation$, 4], 5
];
exports.Classification$ = [3, n0, _C,
    0,
    [_N, _Pr],
    [0, 128 | 0]
];
exports.CloudWatchLoggingConfiguration$ = [3, n0, _CWLC,
    0,
    [_E, _LG, _LSNP, _LT],
    [2, 0, 0, [2, n0, _LTM, 0, 0, 64 | 0]], 1
];
exports.Column$ = [3, n0, _Co,
    0,
    [_N, _T, _Com],
    [0, 0, 0], 1
];
exports.ColumnInfo$ = [3, n0, _CI,
    0,
    [_N, _T, _CN, _SN, _TN, _L, _Pre, _Sc, _Nu, _CSas],
    [0, 0, 0, 0, 0, 0, 1, 1, 0, 2], 2
];
exports.CreateCapacityReservationInput$ = [3, n0, _CCRIr,
    0,
    [_TD, _N, _Ta],
    [1, 0, () => TagList], 2
];
exports.CreateCapacityReservationOutput$ = [3, n0, _CCROr,
    0,
    [],
    []
];
exports.CreateDataCatalogInput$ = [3, n0, _CDCI,
    0,
    [_N, _T, _D, _Pa, _Ta],
    [0, 0, 0, 128 | 0, () => TagList], 2
];
exports.CreateDataCatalogOutput$ = [3, n0, _CDCO,
    0,
    [_DC],
    [() => exports.DataCatalog$]
];
exports.CreateNamedQueryInput$ = [3, n0, _CNQI,
    0,
    [_N, _Da, _QS, _D, _CRT, _WG],
    [0, 0, 0, 0, [0, 4], 0], 3
];
exports.CreateNamedQueryOutput$ = [3, n0, _CNQO,
    0,
    [_NQIa],
    [0]
];
exports.CreateNotebookInput$ = [3, n0, _CNI,
    0,
    [_WG, _N, _CRT],
    [0, 0, 0], 2
];
exports.CreateNotebookOutput$ = [3, n0, _CNO,
    0,
    [_NI],
    [0]
];
exports.CreatePreparedStatementInput$ = [3, n0, _CPSI,
    0,
    [_SNt, _WG, _QSu, _D],
    [0, 0, 0, 0], 3
];
exports.CreatePreparedStatementOutput$ = [3, n0, _CPSO,
    0,
    [],
    []
];
exports.CreatePresignedNotebookUrlRequest$ = [3, n0, _CPNUR,
    0,
    [_SI],
    [0], 1
];
exports.CreatePresignedNotebookUrlResponse$ = [3, n0, _CPNURr,
    0,
    [_NU, _AT, _ATET],
    [0, 0, 1], 3
];
exports.CreateWorkGroupInput$ = [3, n0, _CWGI,
    0,
    [_N, _Con, _D, _Ta],
    [0, () => exports.WorkGroupConfiguration$, 0, () => TagList], 1
];
exports.CreateWorkGroupOutput$ = [3, n0, _CWGO,
    0,
    [],
    []
];
exports.CustomerContentEncryptionConfiguration$ = [3, n0, _CCEC,
    0,
    [_KK],
    [0], 1
];
exports.Database$ = [3, n0, _Da,
    0,
    [_N, _D, _Pa],
    [0, 0, 128 | 0], 1
];
exports.DataCatalog$ = [3, n0, _DC,
    0,
    [_N, _T, _D, _Pa, _St, _CTo, _Er],
    [0, 0, 0, 128 | 0, 0, 0, 0], 2
];
exports.DataCatalogSummary$ = [3, n0, _DCS,
    0,
    [_CN, _T, _St, _CTo, _Er],
    [0, 0, 0, 0, 0]
];
exports.Datum$ = [3, n0, _Dat,
    0,
    [_VCV],
    [0]
];
exports.DeleteCapacityReservationInput$ = [3, n0, _DCRI,
    0,
    [_N],
    [0], 1
];
exports.DeleteCapacityReservationOutput$ = [3, n0, _DCRO,
    0,
    [],
    []
];
exports.DeleteDataCatalogInput$ = [3, n0, _DDCI,
    0,
    [_N, _DCO],
    [0, 2], 1
];
exports.DeleteDataCatalogOutput$ = [3, n0, _DDCO,
    0,
    [_DC],
    [() => exports.DataCatalog$]
];
exports.DeleteNamedQueryInput$ = [3, n0, _DNQI,
    0,
    [_NQIa],
    [[0, 4]]
];
exports.DeleteNamedQueryOutput$ = [3, n0, _DNQO,
    0,
    [],
    []
];
exports.DeleteNotebookInput$ = [3, n0, _DNI,
    0,
    [_NI],
    [0], 1
];
exports.DeleteNotebookOutput$ = [3, n0, _DNO,
    0,
    [],
    []
];
exports.DeletePreparedStatementInput$ = [3, n0, _DPSI,
    0,
    [_SNt, _WG],
    [0, 0], 2
];
exports.DeletePreparedStatementOutput$ = [3, n0, _DPSO,
    0,
    [],
    []
];
exports.DeleteWorkGroupInput$ = [3, n0, _DWGI,
    0,
    [_WG, _RDO],
    [0, 2], 1
];
exports.DeleteWorkGroupOutput$ = [3, n0, _DWGO,
    0,
    [],
    []
];
exports.EncryptionConfiguration$ = [3, n0, _ECn,
    0,
    [_EO, _KK],
    [0, 0], 1
];
exports.EngineConfiguration$ = [3, n0, _ECng,
    0,
    [_CDS, _MCD, _DEDS, _ACd, _SP, _Cl],
    [1, 1, 1, 128 | 0, 128 | 0, () => ClassificationList]
];
exports.EngineVersion$ = [3, n0, _EV,
    0,
    [_SEV, _EEV],
    [0, 0]
];
exports.ExecutorsSummary$ = [3, n0, _ES,
    0,
    [_EI, _ETx, _SDTt, _TDT, _ESx, _ESxe],
    [0, 0, 1, 1, 0, 1], 1
];
exports.ExportNotebookInput$ = [3, n0, _ENI,
    0,
    [_NI],
    [0], 1
];
exports.ExportNotebookOutput$ = [3, n0, _ENO,
    0,
    [_NM, _Pay],
    [() => exports.NotebookMetadata$, 0]
];
exports.FilterDefinition$ = [3, n0, _FD,
    0,
    [_N],
    [0]
];
exports.GetCalculationExecutionCodeRequest$ = [3, n0, _GCECR,
    0,
    [_CEI],
    [0], 1
];
exports.GetCalculationExecutionCodeResponse$ = [3, n0, _GCECRe,
    0,
    [_CB],
    [0]
];
exports.GetCalculationExecutionRequest$ = [3, n0, _GCER,
    0,
    [_CEI],
    [0], 1
];
exports.GetCalculationExecutionResponse$ = [3, n0, _GCERe,
    0,
    [_CEI, _SI, _D, _WD, _St, _Sta, _Res],
    [0, 0, 0, 0, () => exports.CalculationStatus$, () => exports.CalculationStatistics$, () => exports.CalculationResult$]
];
exports.GetCalculationExecutionStatusRequest$ = [3, n0, _GCESR,
    0,
    [_CEI],
    [0], 1
];
exports.GetCalculationExecutionStatusResponse$ = [3, n0, _GCESRe,
    0,
    [_St, _Sta],
    [() => exports.CalculationStatus$, () => exports.CalculationStatistics$]
];
exports.GetCapacityAssignmentConfigurationInput$ = [3, n0, _GCACI,
    0,
    [_CRN],
    [0], 1
];
exports.GetCapacityAssignmentConfigurationOutput$ = [3, n0, _GCACO,
    0,
    [_CAC],
    [() => exports.CapacityAssignmentConfiguration$], 1
];
exports.GetCapacityReservationInput$ = [3, n0, _GCRI,
    0,
    [_N],
    [0], 1
];
exports.GetCapacityReservationOutput$ = [3, n0, _GCRO,
    0,
    [_CRa],
    [() => exports.CapacityReservation$], 1
];
exports.GetDatabaseInput$ = [3, n0, _GDI,
    0,
    [_CN, _DN, _WG],
    [0, 0, 0], 2
];
exports.GetDatabaseOutput$ = [3, n0, _GDO,
    0,
    [_Da],
    [() => exports.Database$]
];
exports.GetDataCatalogInput$ = [3, n0, _GDCI,
    0,
    [_N, _WG],
    [0, 0], 1
];
exports.GetDataCatalogOutput$ = [3, n0, _GDCO,
    0,
    [_DC],
    [() => exports.DataCatalog$]
];
exports.GetNamedQueryInput$ = [3, n0, _GNQI,
    0,
    [_NQIa],
    [0], 1
];
exports.GetNamedQueryOutput$ = [3, n0, _GNQO,
    0,
    [_NQa],
    [() => exports.NamedQuery$]
];
exports.GetNotebookMetadataInput$ = [3, n0, _GNMI,
    0,
    [_NI],
    [0], 1
];
exports.GetNotebookMetadataOutput$ = [3, n0, _GNMO,
    0,
    [_NM],
    [() => exports.NotebookMetadata$]
];
exports.GetPreparedStatementInput$ = [3, n0, _GPSI,
    0,
    [_SNt, _WG],
    [0, 0], 2
];
exports.GetPreparedStatementOutput$ = [3, n0, _GPSO,
    0,
    [_PSr],
    [() => exports.PreparedStatement$]
];
exports.GetQueryExecutionInput$ = [3, n0, _GQEI,
    0,
    [_QEIu],
    [0], 1
];
exports.GetQueryExecutionOutput$ = [3, n0, _GQEO,
    0,
    [_QEu],
    [() => exports.QueryExecution$]
];
exports.GetQueryResultsInput$ = [3, n0, _GQRI,
    0,
    [_QEIu, _NT, _MR, _QRT],
    [0, 0, 1, 0], 1
];
exports.GetQueryResultsOutput$ = [3, n0, _GQRO,
    0,
    [_UC, _RS, _NT],
    [1, () => exports.ResultSet$, 0]
];
exports.GetQueryRuntimeStatisticsInput$ = [3, n0, _GQRSI,
    0,
    [_QEIu],
    [0], 1
];
exports.GetQueryRuntimeStatisticsOutput$ = [3, n0, _GQRSO,
    0,
    [_QRS],
    [() => exports.QueryRuntimeStatistics$]
];
exports.GetResourceDashboardRequest$ = [3, n0, _GRDR,
    0,
    [_RARN],
    [0], 1
];
exports.GetResourceDashboardResponse$ = [3, n0, _GRDRe,
    0,
    [_U],
    [0], 1
];
exports.GetSessionEndpointRequest$ = [3, n0, _GSER,
    0,
    [_SI],
    [0], 1
];
exports.GetSessionEndpointResponse$ = [3, n0, _GSERe,
    0,
    [_EU, _AT, _ATET],
    [0, 0, 4], 3
];
exports.GetSessionRequest$ = [3, n0, _GSR,
    0,
    [_SI],
    [0], 1
];
exports.GetSessionResponse$ = [3, n0, _GSRe,
    0,
    [_SI, _D, _WG, _EV, _ECng, _NV, _MC, _SC, _St, _Sta],
    [0, 0, 0, 0, () => exports.EngineConfiguration$, 0, () => exports.MonitoringConfiguration$, () => exports.SessionConfiguration$, () => exports.SessionStatus$, () => exports.SessionStatistics$]
];
exports.GetSessionStatusRequest$ = [3, n0, _GSSR,
    0,
    [_SI],
    [0], 1
];
exports.GetSessionStatusResponse$ = [3, n0, _GSSRe,
    0,
    [_SI, _St],
    [0, () => exports.SessionStatus$]
];
exports.GetTableMetadataInput$ = [3, n0, _GTMI,
    0,
    [_CN, _DN, _TN, _WG],
    [0, 0, 0, 0], 3
];
exports.GetTableMetadataOutput$ = [3, n0, _GTMO,
    0,
    [_TM],
    [() => exports.TableMetadata$]
];
exports.GetWorkGroupInput$ = [3, n0, _GWGI,
    0,
    [_WG],
    [0], 1
];
exports.GetWorkGroupOutput$ = [3, n0, _GWGO,
    0,
    [_WG],
    [() => exports.WorkGroup$]
];
exports.IdentityCenterConfiguration$ = [3, n0, _ICC,
    0,
    [_EIC, _ICIA],
    [2, 0]
];
exports.ImportNotebookInput$ = [3, n0, _INI,
    0,
    [_WG, _N, _T, _Pay, _NSLU, _CRT],
    [0, 0, 0, 0, 0, 0], 3
];
exports.ImportNotebookOutput$ = [3, n0, _INO,
    0,
    [_NI],
    [0]
];
exports.ListApplicationDPUSizesInput$ = [3, n0, _LADPUSI,
    0,
    [_MR, _NT],
    [1, 0]
];
exports.ListApplicationDPUSizesOutput$ = [3, n0, _LADPUSO,
    0,
    [_ADPUS, _NT],
    [() => ApplicationDPUSizesList, 0]
];
exports.ListCalculationExecutionsRequest$ = [3, n0, _LCER,
    0,
    [_SI, _SF, _MR, _NT],
    [0, 0, 1, 0], 1
];
exports.ListCalculationExecutionsResponse$ = [3, n0, _LCERi,
    0,
    [_NT, _Ca],
    [0, () => CalculationsList]
];
exports.ListCapacityReservationsInput$ = [3, n0, _LCRI,
    0,
    [_NT, _MR],
    [0, 1]
];
exports.ListCapacityReservationsOutput$ = [3, n0, _LCRO,
    0,
    [_CRap, _NT],
    [() => CapacityReservationsList, 0], 1
];
exports.ListDatabasesInput$ = [3, n0, _LDI,
    0,
    [_CN, _NT, _MR, _WG],
    [0, 0, 1, 0], 1
];
exports.ListDatabasesOutput$ = [3, n0, _LDO,
    0,
    [_DL, _NT],
    [() => DatabaseList, 0]
];
exports.ListDataCatalogsInput$ = [3, n0, _LDCI,
    0,
    [_NT, _MR, _WG],
    [0, 1, 0]
];
exports.ListDataCatalogsOutput$ = [3, n0, _LDCO,
    0,
    [_DCSa, _NT],
    [() => DataCatalogSummaryList, 0]
];
exports.ListEngineVersionsInput$ = [3, n0, _LEVI,
    0,
    [_NT, _MR],
    [0, 1]
];
exports.ListEngineVersionsOutput$ = [3, n0, _LEVO,
    0,
    [_EVn, _NT],
    [() => EngineVersionsList, 0]
];
exports.ListExecutorsRequest$ = [3, n0, _LER,
    0,
    [_SI, _ESF, _MR, _NT],
    [0, 0, 1, 0], 1
];
exports.ListExecutorsResponse$ = [3, n0, _LERi,
    0,
    [_SI, _NT, _ES],
    [0, 0, () => ExecutorsSummaryList], 1
];
exports.ListNamedQueriesInput$ = [3, n0, _LNQI,
    0,
    [_NT, _MR, _WG],
    [0, 1, 0]
];
exports.ListNamedQueriesOutput$ = [3, n0, _LNQO,
    0,
    [_NQI, _NT],
    [64 | 0, 0]
];
exports.ListNotebookMetadataInput$ = [3, n0, _LNMI,
    0,
    [_WG, _F, _NT, _MR],
    [0, () => exports.FilterDefinition$, 0, 1], 1
];
exports.ListNotebookMetadataOutput$ = [3, n0, _LNMO,
    0,
    [_NT, _NML],
    [0, () => NotebookMetadataArray]
];
exports.ListNotebookSessionsRequest$ = [3, n0, _LNSR,
    0,
    [_NI, _MR, _NT],
    [0, 1, 0], 1
];
exports.ListNotebookSessionsResponse$ = [3, n0, _LNSRi,
    0,
    [_NSL, _NT],
    [() => NotebookSessionsList, 0], 1
];
exports.ListPreparedStatementsInput$ = [3, n0, _LPSI,
    0,
    [_WG, _NT, _MR],
    [0, 0, 1], 1
];
exports.ListPreparedStatementsOutput$ = [3, n0, _LPSO,
    0,
    [_PS, _NT],
    [() => PreparedStatementsList, 0]
];
exports.ListQueryExecutionsInput$ = [3, n0, _LQEI,
    0,
    [_NT, _MR, _WG],
    [0, 1, 0]
];
exports.ListQueryExecutionsOutput$ = [3, n0, _LQEO,
    0,
    [_QEI, _NT],
    [64 | 0, 0]
];
exports.ListSessionsRequest$ = [3, n0, _LSR,
    0,
    [_WG, _SF, _MR, _NT],
    [0, 0, 1, 0], 1
];
exports.ListSessionsResponse$ = [3, n0, _LSRi,
    0,
    [_NT, _Se],
    [0, () => SessionsList]
];
exports.ListTableMetadataInput$ = [3, n0, _LTMI,
    0,
    [_CN, _DN, _Ex, _NT, _MR, _WG],
    [0, 0, 0, 0, 1, 0], 2
];
exports.ListTableMetadataOutput$ = [3, n0, _LTMO,
    0,
    [_TML, _NT],
    [() => TableMetadataList, 0]
];
exports.ListTagsForResourceInput$ = [3, n0, _LTFRI,
    0,
    [_RARN, _NT, _MR],
    [0, 0, 1], 1
];
exports.ListTagsForResourceOutput$ = [3, n0, _LTFRO,
    0,
    [_Ta, _NT],
    [() => TagList, 0]
];
exports.ListWorkGroupsInput$ = [3, n0, _LWGI,
    0,
    [_NT, _MR],
    [0, 1]
];
exports.ListWorkGroupsOutput$ = [3, n0, _LWGO,
    0,
    [_WGo, _NT],
    [() => WorkGroupsList, 0]
];
exports.ManagedLoggingConfiguration$ = [3, n0, _MLC,
    0,
    [_E, _KK],
    [2, 0], 1
];
exports.ManagedQueryResultsConfiguration$ = [3, n0, _MQRC,
    0,
    [_E, _ECn],
    [2, () => exports.ManagedQueryResultsEncryptionConfiguration$], 1
];
exports.ManagedQueryResultsConfigurationUpdates$ = [3, n0, _MQRCU,
    0,
    [_E, _ECn, _REC],
    [2, () => exports.ManagedQueryResultsEncryptionConfiguration$, 2]
];
exports.ManagedQueryResultsEncryptionConfiguration$ = [3, n0, _MQREC,
    0,
    [_KK],
    [0], 1
];
exports.MonitoringConfiguration$ = [3, n0, _MC,
    0,
    [_CWLC, _MLC, _SLC],
    [() => exports.CloudWatchLoggingConfiguration$, () => exports.ManagedLoggingConfiguration$, () => exports.S3LoggingConfiguration$]
];
exports.NamedQuery$ = [3, n0, _NQa,
    0,
    [_N, _Da, _QS, _D, _NQIa, _WG],
    [0, 0, 0, 0, 0, 0], 3
];
exports.NotebookMetadata$ = [3, n0, _NM,
    0,
    [_NI, _N, _WG, _CT, _T, _LMT],
    [0, 0, 0, 4, 0, 4]
];
exports.NotebookSessionSummary$ = [3, n0, _NSS,
    0,
    [_SI, _CT],
    [0, 4]
];
exports.PreparedStatement$ = [3, n0, _PSr,
    0,
    [_SNt, _QSu, _WGNo, _D, _LMT],
    [0, 0, 0, 0, 4]
];
exports.PreparedStatementSummary$ = [3, n0, _PSS,
    0,
    [_SNt, _LMT],
    [0, 4]
];
exports.PutCapacityAssignmentConfigurationInput$ = [3, n0, _PCACI,
    0,
    [_CRN, _CAap],
    [0, () => CapacityAssignmentsList], 2
];
exports.PutCapacityAssignmentConfigurationOutput$ = [3, n0, _PCACO,
    0,
    [],
    []
];
exports.QueryExecution$ = [3, n0, _QEu,
    0,
    [_QEIu, _Q, _ST, _MQRC, _RC, _RRC, _QEC, _St, _Sta, _WG, _EV, _EP, _STu, _QRSAGC],
    [0, 0, 0, () => exports.ManagedQueryResultsConfiguration$, () => exports.ResultConfiguration$, () => exports.ResultReuseConfiguration$, () => exports.QueryExecutionContext$, () => exports.QueryExecutionStatus$, () => exports.QueryExecutionStatistics$, 0, () => exports.EngineVersion$, 64 | 0, 0, () => exports.QueryResultsS3AccessGrantsConfiguration$]
];
exports.QueryExecutionContext$ = [3, n0, _QEC,
    0,
    [_Da, _Cat],
    [0, 0]
];
exports.QueryExecutionStatistics$ = [3, n0, _QES,
    0,
    [_EETIM, _DSIB, _DML, _TETIM, _QQTIM, _SPPTIM, _QPTIM, _SPTIM, _RRI, _DCp],
    [1, 1, 0, 1, 1, 1, 1, 1, () => exports.ResultReuseInformation$, 1]
];
exports.QueryExecutionStatus$ = [3, n0, _QESu,
    0,
    [_S, _SCR, _SDT, _CDT, _AE],
    [0, 0, 4, 4, () => exports.AthenaError$]
];
exports.QueryResultsS3AccessGrantsConfiguration$ = [3, n0, _QRSAGC,
    0,
    [_ESAG, _ATu, _CULP],
    [2, 0, 2], 2
];
exports.QueryRuntimeStatistics$ = [3, n0, _QRS,
    0,
    [_Ti, _Ro, _OS],
    [() => exports.QueryRuntimeStatisticsTimeline$, () => exports.QueryRuntimeStatisticsRows$, () => exports.QueryStage$]
];
exports.QueryRuntimeStatisticsRows$ = [3, n0, _QRSR,
    0,
    [_IR, _IB, _OB, _OR],
    [1, 1, 1, 1]
];
exports.QueryRuntimeStatisticsTimeline$ = [3, n0, _QRST,
    0,
    [_QQTIM, _SPPTIM, _QPTIM, _EETIM, _SPTIM, _TETIM],
    [1, 1, 1, 1, 1, 1]
];
exports.QueryStage$ = [3, n0, _QSue,
    0,
    [_SIt, _S, _OB, _OR, _IB, _IR, _ETxe, _QSP, _SS],
    [1, 0, 1, 1, 1, 1, 1, () => exports.QueryStagePlanNode$, () => QueryStages]
];
exports.QueryStagePlanNode$ = [3, n0, _QSPN,
    0,
    [_N, _I, _Ch, _RSe],
    [0, 0, () => QueryStagePlanNodes, 64 | 0]
];
exports.ResultConfiguration$ = [3, n0, _RC,
    0,
    [_OL, _ECn, _EBO, _AC],
    [0, () => exports.EncryptionConfiguration$, 0, () => exports.AclConfiguration$]
];
exports.ResultConfigurationUpdates$ = [3, n0, _RCU,
    0,
    [_OL, _ROL, _ECn, _REC, _EBO, _REBO, _AC, _RAC],
    [0, 2, () => exports.EncryptionConfiguration$, 2, 0, 2, () => exports.AclConfiguration$, 2]
];
exports.ResultReuseByAgeConfiguration$ = [3, n0, _RRBAC,
    0,
    [_E, _MAIM],
    [2, 1], 1
];
exports.ResultReuseConfiguration$ = [3, n0, _RRC,
    0,
    [_RRBAC],
    [() => exports.ResultReuseByAgeConfiguration$]
];
exports.ResultReuseInformation$ = [3, n0, _RRI,
    0,
    [_RPR],
    [2], 1
];
exports.ResultSet$ = [3, n0, _RS,
    0,
    [_Ro, _RSM],
    [() => RowList, () => exports.ResultSetMetadata$]
];
exports.ResultSetMetadata$ = [3, n0, _RSM,
    0,
    [_CI],
    [() => ColumnInfoList]
];
exports.Row$ = [3, n0, _Row,
    0,
    [_Data],
    [() => datumList]
];
exports.S3LoggingConfiguration$ = [3, n0, _SLC,
    0,
    [_E, _KK, _LL],
    [2, 0, 0], 1
];
exports.SessionConfiguration$ = [3, n0, _SC,
    0,
    [_ER, _WD, _ITS, _SITIM, _ECn],
    [0, 0, 1, 1, () => exports.EncryptionConfiguration$]
];
exports.SessionStatistics$ = [3, n0, _SSe,
    0,
    [_DEIM],
    [1]
];
exports.SessionStatus$ = [3, n0, _SSes,
    0,
    [_SDTt, _LMDT, _EDT, _ISDT, _S, _SCR],
    [4, 4, 4, 4, 0, 0]
];
exports.SessionSummary$ = [3, n0, _SSess,
    0,
    [_SI, _D, _EV, _NV, _St],
    [0, 0, () => exports.EngineVersion$, 0, () => exports.SessionStatus$]
];
exports.StartCalculationExecutionRequest$ = [3, n0, _SCER,
    0,
    [_SI, _D, _CC, _CB, _CRT],
    [0, 0, () => exports.CalculationConfiguration$, 0, 0], 1
];
exports.StartCalculationExecutionResponse$ = [3, n0, _SCERt,
    0,
    [_CEI, _S],
    [0, 0]
];
exports.StartQueryExecutionInput$ = [3, n0, _SQEI,
    0,
    [_QS, _CRT, _QEC, _RC, _WG, _EP, _RRC, _ECng],
    [0, [0, 4], () => exports.QueryExecutionContext$, () => exports.ResultConfiguration$, 0, 64 | 0, () => exports.ResultReuseConfiguration$, () => exports.EngineConfiguration$], 1
];
exports.StartQueryExecutionOutput$ = [3, n0, _SQEO,
    0,
    [_QEIu],
    [0]
];
exports.StartSessionRequest$ = [3, n0, _SSR,
    0,
    [_WG, _ECng, _D, _ER, _MC, _NV, _SITIM, _CRT, _Ta, _CWGT],
    [0, () => exports.EngineConfiguration$, 0, 0, () => exports.MonitoringConfiguration$, 0, 1, 0, () => TagList, 2], 2
];
exports.StartSessionResponse$ = [3, n0, _SSRt,
    0,
    [_SI, _S],
    [0, 0]
];
exports.StopCalculationExecutionRequest$ = [3, n0, _SCERto,
    0,
    [_CEI],
    [0], 1
];
exports.StopCalculationExecutionResponse$ = [3, n0, _SCERtop,
    0,
    [_S],
    [0]
];
exports.StopQueryExecutionInput$ = [3, n0, _SQEIt,
    0,
    [_QEIu],
    [[0, 4]]
];
exports.StopQueryExecutionOutput$ = [3, n0, _SQEOt,
    0,
    [],
    []
];
exports.TableMetadata$ = [3, n0, _TM,
    0,
    [_N, _CTr, _LAT, _TT, _Col, _PK, _Pa],
    [0, 4, 4, 0, () => ColumnList, () => ColumnList, 128 | 0], 1
];
exports.Tag$ = [3, n0, _Tag,
    0,
    [_K, _V],
    [0, 0]
];
exports.TagResourceInput$ = [3, n0, _TRI,
    0,
    [_RARN, _Ta],
    [0, () => TagList], 2
];
exports.TagResourceOutput$ = [3, n0, _TRO,
    0,
    [],
    []
];
exports.TerminateSessionRequest$ = [3, n0, _TSR,
    0,
    [_SI],
    [0], 1
];
exports.TerminateSessionResponse$ = [3, n0, _TSRe,
    0,
    [_S],
    [0]
];
exports.UnprocessedNamedQueryId$ = [3, n0, _UNQIn,
    0,
    [_NQIa, _ECr, _EM],
    [0, 0, 0]
];
exports.UnprocessedPreparedStatementName$ = [3, n0, _UPSNn,
    0,
    [_SNt, _ECr, _EM],
    [0, 0, 0]
];
exports.UnprocessedQueryExecutionId$ = [3, n0, _UQEIn,
    0,
    [_QEIu, _ECr, _EM],
    [0, 0, 0]
];
exports.UntagResourceInput$ = [3, n0, _URI,
    0,
    [_RARN, _TK],
    [0, 64 | 0], 2
];
exports.UntagResourceOutput$ = [3, n0, _URO,
    0,
    [],
    []
];
exports.UpdateCapacityReservationInput$ = [3, n0, _UCRI,
    0,
    [_TD, _N],
    [1, 0], 2
];
exports.UpdateCapacityReservationOutput$ = [3, n0, _UCRO,
    0,
    [],
    []
];
exports.UpdateDataCatalogInput$ = [3, n0, _UDCI,
    0,
    [_N, _T, _D, _Pa],
    [0, 0, 0, 128 | 0], 2
];
exports.UpdateDataCatalogOutput$ = [3, n0, _UDCO,
    0,
    [],
    []
];
exports.UpdateNamedQueryInput$ = [3, n0, _UNQIp,
    0,
    [_NQIa, _N, _QS, _D],
    [0, 0, 0, 0], 3
];
exports.UpdateNamedQueryOutput$ = [3, n0, _UNQO,
    0,
    [],
    []
];
exports.UpdateNotebookInput$ = [3, n0, _UNI,
    0,
    [_NI, _Pay, _T, _SI, _CRT],
    [0, 0, 0, 0, 0], 3
];
exports.UpdateNotebookMetadataInput$ = [3, n0, _UNMI,
    0,
    [_NI, _N, _CRT],
    [0, 0, 0], 2
];
exports.UpdateNotebookMetadataOutput$ = [3, n0, _UNMO,
    0,
    [],
    []
];
exports.UpdateNotebookOutput$ = [3, n0, _UNO,
    0,
    [],
    []
];
exports.UpdatePreparedStatementInput$ = [3, n0, _UPSI,
    0,
    [_SNt, _WG, _QSu, _D],
    [0, 0, 0, 0], 3
];
exports.UpdatePreparedStatementOutput$ = [3, n0, _UPSO,
    0,
    [],
    []
];
exports.UpdateWorkGroupInput$ = [3, n0, _UWGI,
    0,
    [_WG, _D, _CU, _S],
    [0, 0, () => exports.WorkGroupConfigurationUpdates$, 0], 1
];
exports.UpdateWorkGroupOutput$ = [3, n0, _UWGO,
    0,
    [],
    []
];
exports.WorkGroup$ = [3, n0, _WG,
    0,
    [_N, _S, _Con, _D, _CT, _ICAA],
    [0, 0, () => exports.WorkGroupConfiguration$, 0, 4, 0], 1
];
exports.WorkGroupConfiguration$ = [3, n0, _WGC,
    0,
    [_RC, _MQRC, _EWGC, _PCWME, _BSCPQ, _RPE, _EV, _ACdd, _ER, _MC, _ECng, _CCEC, _EMEC, _ICC, _QRSAGC],
    [() => exports.ResultConfiguration$, () => exports.ManagedQueryResultsConfiguration$, 2, 2, 1, 2, () => exports.EngineVersion$, 0, 0, () => exports.MonitoringConfiguration$, () => exports.EngineConfiguration$, () => exports.CustomerContentEncryptionConfiguration$, 2, () => exports.IdentityCenterConfiguration$, () => exports.QueryResultsS3AccessGrantsConfiguration$]
];
exports.WorkGroupConfigurationUpdates$ = [3, n0, _WGCU,
    0,
    [_EWGC, _RCU, _MQRCU, _PCWME, _BSCPQ, _RBSCPQ, _RPE, _EV, _RCCEC, _ACdd, _ER, _CCEC, _EMEC, _QRSAGC, _MC, _ECng],
    [2, () => exports.ResultConfigurationUpdates$, () => exports.ManagedQueryResultsConfigurationUpdates$, 2, 1, 2, 2, () => exports.EngineVersion$, 2, 0, 0, () => exports.CustomerContentEncryptionConfiguration$, 2, () => exports.QueryResultsS3AccessGrantsConfiguration$, () => exports.MonitoringConfiguration$, () => exports.EngineConfiguration$]
];
exports.WorkGroupSummary$ = [3, n0, _WGS,
    0,
    [_N, _S, _D, _CT, _EV, _ICAA],
    [0, 0, 0, 4, () => exports.EngineVersion$, 0]
];
var ApplicationDPUSizesList = [1, n0, _ADPUSL,
    0, () => exports.ApplicationDPUSizes$
];
var CalculationsList = [1, n0, _CL,
    0, () => exports.CalculationSummary$
];
var CapacityAssignmentsList = [1, n0, _CAL,
    0, () => exports.CapacityAssignment$
];
var CapacityReservationsList = [1, n0, _CRL,
    0, () => exports.CapacityReservation$
];
var ClassificationList = [1, n0, _CLl,
    0, () => exports.Classification$
];
var ColumnInfoList = [1, n0, _CIL,
    0, () => exports.ColumnInfo$
];
var ColumnList = [1, n0, _CLo,
    0, () => exports.Column$
];
var DatabaseList = [1, n0, _DL,
    0, () => exports.Database$
];
var DataCatalogSummaryList = [1, n0, _DCSL,
    0, () => exports.DataCatalogSummary$
];
var datumList = [1, n0, _dL,
    0, () => exports.Datum$
];
var EngineVersionsList = [1, n0, _EVL,
    0, () => exports.EngineVersion$
];
var ExecutionParameters = 64 | 0;
var ExecutorsSummaryList = [1, n0, _ESL,
    0, () => exports.ExecutorsSummary$
];
var LogTypeValuesList = 64 | 0;
var NamedQueryIdList = 64 | 0;
var NamedQueryList = [1, n0, _NQL,
    0, () => exports.NamedQuery$
];
var NotebookMetadataArray = [1, n0, _NMA,
    0, () => exports.NotebookMetadata$
];
var NotebookSessionsList = [1, n0, _NSL,
    0, () => exports.NotebookSessionSummary$
];
var PreparedStatementDetailsList = [1, n0, _PSDL,
    0, () => exports.PreparedStatement$
];
var PreparedStatementNameList = 64 | 0;
var PreparedStatementsList = [1, n0, _PSL,
    0, () => exports.PreparedStatementSummary$
];
var QueryExecutionIdList = 64 | 0;
var QueryExecutionList = [1, n0, _QEL,
    0, () => exports.QueryExecution$
];
var QueryStagePlanNodes = [1, n0, _QSPNu,
    0, () => exports.QueryStagePlanNode$
];
var QueryStages = [1, n0, _QSuer,
    0, () => exports.QueryStage$
];
var RowList = [1, n0, _RL,
    0, () => exports.Row$
];
var SessionsList = [1, n0, _SL,
    0, () => exports.SessionSummary$
];
var StringList = 64 | 0;
var SupportedDPUSizeList = 64 | 1;
var TableMetadataList = [1, n0, _TML,
    0, () => exports.TableMetadata$
];
var TagKeyList = 64 | 0;
var TagList = [1, n0, _TL,
    0, () => exports.Tag$
];
var UnprocessedNamedQueryIdList = [1, n0, _UNQIL,
    0, () => exports.UnprocessedNamedQueryId$
];
var UnprocessedPreparedStatementNameList = [1, n0, _UPSNL,
    0, () => exports.UnprocessedPreparedStatementName$
];
var UnprocessedQueryExecutionIdList = [1, n0, _UQEIL,
    0, () => exports.UnprocessedQueryExecutionId$
];
var WorkGroupNamesList = 64 | 0;
var WorkGroupsList = [1, n0, _WGL,
    0, () => exports.WorkGroupSummary$
];
var LogTypesMap = [2, n0, _LTM,
    0, 0, 64 | 0
];
var ParametersMap = 128 | 0;
exports.BatchGetNamedQuery$ = [9, n0, _BGNQ,
    0, () => exports.BatchGetNamedQueryInput$, () => exports.BatchGetNamedQueryOutput$
];
exports.BatchGetPreparedStatement$ = [9, n0, _BGPS,
    0, () => exports.BatchGetPreparedStatementInput$, () => exports.BatchGetPreparedStatementOutput$
];
exports.BatchGetQueryExecution$ = [9, n0, _BGQE,
    0, () => exports.BatchGetQueryExecutionInput$, () => exports.BatchGetQueryExecutionOutput$
];
exports.CancelCapacityReservation$ = [9, n0, _CCR,
    2, () => exports.CancelCapacityReservationInput$, () => exports.CancelCapacityReservationOutput$
];
exports.CreateCapacityReservation$ = [9, n0, _CCRr,
    2, () => exports.CreateCapacityReservationInput$, () => exports.CreateCapacityReservationOutput$
];
exports.CreateDataCatalog$ = [9, n0, _CDC,
    0, () => exports.CreateDataCatalogInput$, () => exports.CreateDataCatalogOutput$
];
exports.CreateNamedQuery$ = [9, n0, _CNQ,
    2, () => exports.CreateNamedQueryInput$, () => exports.CreateNamedQueryOutput$
];
exports.CreateNotebook$ = [9, n0, _CNr,
    0, () => exports.CreateNotebookInput$, () => exports.CreateNotebookOutput$
];
exports.CreatePreparedStatement$ = [9, n0, _CPS,
    0, () => exports.CreatePreparedStatementInput$, () => exports.CreatePreparedStatementOutput$
];
exports.CreatePresignedNotebookUrl$ = [9, n0, _CPNU,
    0, () => exports.CreatePresignedNotebookUrlRequest$, () => exports.CreatePresignedNotebookUrlResponse$
];
exports.CreateWorkGroup$ = [9, n0, _CWG,
    0, () => exports.CreateWorkGroupInput$, () => exports.CreateWorkGroupOutput$
];
exports.DeleteCapacityReservation$ = [9, n0, _DCR,
    2, () => exports.DeleteCapacityReservationInput$, () => exports.DeleteCapacityReservationOutput$
];
exports.DeleteDataCatalog$ = [9, n0, _DDC,
    0, () => exports.DeleteDataCatalogInput$, () => exports.DeleteDataCatalogOutput$
];
exports.DeleteNamedQuery$ = [9, n0, _DNQ,
    2, () => exports.DeleteNamedQueryInput$, () => exports.DeleteNamedQueryOutput$
];
exports.DeleteNotebook$ = [9, n0, _DNe,
    0, () => exports.DeleteNotebookInput$, () => exports.DeleteNotebookOutput$
];
exports.DeletePreparedStatement$ = [9, n0, _DPS,
    0, () => exports.DeletePreparedStatementInput$, () => exports.DeletePreparedStatementOutput$
];
exports.DeleteWorkGroup$ = [9, n0, _DWG,
    2, () => exports.DeleteWorkGroupInput$, () => exports.DeleteWorkGroupOutput$
];
exports.ExportNotebook$ = [9, n0, _EN,
    0, () => exports.ExportNotebookInput$, () => exports.ExportNotebookOutput$
];
exports.GetCalculationExecution$ = [9, n0, _GCE,
    0, () => exports.GetCalculationExecutionRequest$, () => exports.GetCalculationExecutionResponse$
];
exports.GetCalculationExecutionCode$ = [9, n0, _GCEC,
    0, () => exports.GetCalculationExecutionCodeRequest$, () => exports.GetCalculationExecutionCodeResponse$
];
exports.GetCalculationExecutionStatus$ = [9, n0, _GCES,
    0, () => exports.GetCalculationExecutionStatusRequest$, () => exports.GetCalculationExecutionStatusResponse$
];
exports.GetCapacityAssignmentConfiguration$ = [9, n0, _GCAC,
    0, () => exports.GetCapacityAssignmentConfigurationInput$, () => exports.GetCapacityAssignmentConfigurationOutput$
];
exports.GetCapacityReservation$ = [9, n0, _GCR,
    0, () => exports.GetCapacityReservationInput$, () => exports.GetCapacityReservationOutput$
];
exports.GetDatabase$ = [9, n0, _GD,
    0, () => exports.GetDatabaseInput$, () => exports.GetDatabaseOutput$
];
exports.GetDataCatalog$ = [9, n0, _GDC,
    0, () => exports.GetDataCatalogInput$, () => exports.GetDataCatalogOutput$
];
exports.GetNamedQuery$ = [9, n0, _GNQ,
    0, () => exports.GetNamedQueryInput$, () => exports.GetNamedQueryOutput$
];
exports.GetNotebookMetadata$ = [9, n0, _GNM,
    0, () => exports.GetNotebookMetadataInput$, () => exports.GetNotebookMetadataOutput$
];
exports.GetPreparedStatement$ = [9, n0, _GPS,
    0, () => exports.GetPreparedStatementInput$, () => exports.GetPreparedStatementOutput$
];
exports.GetQueryExecution$ = [9, n0, _GQE,
    0, () => exports.GetQueryExecutionInput$, () => exports.GetQueryExecutionOutput$
];
exports.GetQueryResults$ = [9, n0, _GQR,
    0, () => exports.GetQueryResultsInput$, () => exports.GetQueryResultsOutput$
];
exports.GetQueryRuntimeStatistics$ = [9, n0, _GQRS,
    0, () => exports.GetQueryRuntimeStatisticsInput$, () => exports.GetQueryRuntimeStatisticsOutput$
];
exports.GetResourceDashboard$ = [9, n0, _GRD,
    0, () => exports.GetResourceDashboardRequest$, () => exports.GetResourceDashboardResponse$
];
exports.GetSession$ = [9, n0, _GS,
    0, () => exports.GetSessionRequest$, () => exports.GetSessionResponse$
];
exports.GetSessionEndpoint$ = [9, n0, _GSE,
    0, () => exports.GetSessionEndpointRequest$, () => exports.GetSessionEndpointResponse$
];
exports.GetSessionStatus$ = [9, n0, _GSS,
    0, () => exports.GetSessionStatusRequest$, () => exports.GetSessionStatusResponse$
];
exports.GetTableMetadata$ = [9, n0, _GTM,
    0, () => exports.GetTableMetadataInput$, () => exports.GetTableMetadataOutput$
];
exports.GetWorkGroup$ = [9, n0, _GWG,
    0, () => exports.GetWorkGroupInput$, () => exports.GetWorkGroupOutput$
];
exports.ImportNotebook$ = [9, n0, _IN,
    0, () => exports.ImportNotebookInput$, () => exports.ImportNotebookOutput$
];
exports.ListApplicationDPUSizes$ = [9, n0, _LADPUS,
    0, () => exports.ListApplicationDPUSizesInput$, () => exports.ListApplicationDPUSizesOutput$
];
exports.ListCalculationExecutions$ = [9, n0, _LCE,
    0, () => exports.ListCalculationExecutionsRequest$, () => exports.ListCalculationExecutionsResponse$
];
exports.ListCapacityReservations$ = [9, n0, _LCR,
    0, () => exports.ListCapacityReservationsInput$, () => exports.ListCapacityReservationsOutput$
];
exports.ListDatabases$ = [9, n0, _LD,
    0, () => exports.ListDatabasesInput$, () => exports.ListDatabasesOutput$
];
exports.ListDataCatalogs$ = [9, n0, _LDC,
    0, () => exports.ListDataCatalogsInput$, () => exports.ListDataCatalogsOutput$
];
exports.ListEngineVersions$ = [9, n0, _LEV,
    0, () => exports.ListEngineVersionsInput$, () => exports.ListEngineVersionsOutput$
];
exports.ListExecutors$ = [9, n0, _LE,
    0, () => exports.ListExecutorsRequest$, () => exports.ListExecutorsResponse$
];
exports.ListNamedQueries$ = [9, n0, _LNQ,
    0, () => exports.ListNamedQueriesInput$, () => exports.ListNamedQueriesOutput$
];
exports.ListNotebookMetadata$ = [9, n0, _LNM,
    0, () => exports.ListNotebookMetadataInput$, () => exports.ListNotebookMetadataOutput$
];
exports.ListNotebookSessions$ = [9, n0, _LNS,
    0, () => exports.ListNotebookSessionsRequest$, () => exports.ListNotebookSessionsResponse$
];
exports.ListPreparedStatements$ = [9, n0, _LPS,
    0, () => exports.ListPreparedStatementsInput$, () => exports.ListPreparedStatementsOutput$
];
exports.ListQueryExecutions$ = [9, n0, _LQE,
    0, () => exports.ListQueryExecutionsInput$, () => exports.ListQueryExecutionsOutput$
];
exports.ListSessions$ = [9, n0, _LS,
    0, () => exports.ListSessionsRequest$, () => exports.ListSessionsResponse$
];
exports.ListTableMetadata$ = [9, n0, _LTMi,
    0, () => exports.ListTableMetadataInput$, () => exports.ListTableMetadataOutput$
];
exports.ListTagsForResource$ = [9, n0, _LTFR,
    0, () => exports.ListTagsForResourceInput$, () => exports.ListTagsForResourceOutput$
];
exports.ListWorkGroups$ = [9, n0, _LWG,
    0, () => exports.ListWorkGroupsInput$, () => exports.ListWorkGroupsOutput$
];
exports.PutCapacityAssignmentConfiguration$ = [9, n0, _PCAC,
    2, () => exports.PutCapacityAssignmentConfigurationInput$, () => exports.PutCapacityAssignmentConfigurationOutput$
];
exports.StartCalculationExecution$ = [9, n0, _SCE,
    0, () => exports.StartCalculationExecutionRequest$, () => exports.StartCalculationExecutionResponse$
];
exports.StartQueryExecution$ = [9, n0, _SQE,
    2, () => exports.StartQueryExecutionInput$, () => exports.StartQueryExecutionOutput$
];
exports.StartSession$ = [9, n0, _SSt,
    0, () => exports.StartSessionRequest$, () => exports.StartSessionResponse$
];
exports.StopCalculationExecution$ = [9, n0, _SCEt,
    0, () => exports.StopCalculationExecutionRequest$, () => exports.StopCalculationExecutionResponse$
];
exports.StopQueryExecution$ = [9, n0, _SQEt,
    2, () => exports.StopQueryExecutionInput$, () => exports.StopQueryExecutionOutput$
];
exports.TagResource$ = [9, n0, _TR,
    0, () => exports.TagResourceInput$, () => exports.TagResourceOutput$
];
exports.TerminateSession$ = [9, n0, _TS,
    0, () => exports.TerminateSessionRequest$, () => exports.TerminateSessionResponse$
];
exports.UntagResource$ = [9, n0, _UR,
    0, () => exports.UntagResourceInput$, () => exports.UntagResourceOutput$
];
exports.UpdateCapacityReservation$ = [9, n0, _UCR,
    0, () => exports.UpdateCapacityReservationInput$, () => exports.UpdateCapacityReservationOutput$
];
exports.UpdateDataCatalog$ = [9, n0, _UDC,
    0, () => exports.UpdateDataCatalogInput$, () => exports.UpdateDataCatalogOutput$
];
exports.UpdateNamedQuery$ = [9, n0, _UNQ,
    2, () => exports.UpdateNamedQueryInput$, () => exports.UpdateNamedQueryOutput$
];
exports.UpdateNotebook$ = [9, n0, _UN,
    0, () => exports.UpdateNotebookInput$, () => exports.UpdateNotebookOutput$
];
exports.UpdateNotebookMetadata$ = [9, n0, _UNM,
    0, () => exports.UpdateNotebookMetadataInput$, () => exports.UpdateNotebookMetadataOutput$
];
exports.UpdatePreparedStatement$ = [9, n0, _UPS,
    0, () => exports.UpdatePreparedStatementInput$, () => exports.UpdatePreparedStatementOutput$
];
exports.UpdateWorkGroup$ = [9, n0, _UWG,
    0, () => exports.UpdateWorkGroupInput$, () => exports.UpdateWorkGroupOutput$
];
